[Singh Abdullah Alexander] iP

alexandermula opened the pull request:
### Chacha

> “If you fail to plan, you are planning to fail.” – Benjamin Franklin [(view quote)](https://www.qlik.com/blog/if-you-fail-to-plan-you-are-planning-to-fail-benjamin-franklin#:~:text=Subscribe%20Now-,%E2%80%9CIf%20You%20Fail%20to%20Plan%2C%20You%20Are,Planning%20to%20Fail%E2%80%9D%20%E2%80%94%20Benjamin%20Franklin)

Chacha helps you plan an track upcoming tasks. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/admin/ip-w4.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :blush:

And it is **FREE**!

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [x] Managing events
- [x] Sorting tasks
- [ ] Reminders (coming soon)
---
If you are familiar with Java, you can use it to practice too. Here's the `main` method:
```java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Commit:
Add Gradle support

alexandermula committed:
Completed Level-1

alexandermula committed:
added list

alexandermula committed:
Added mark as done

alexandermula committed:
ToDos, Events and Deadlines

alexandermula committed:
Text testing

alexandermula committed:
error handling

alexandermula committed:
Implemented delete

alexandermula committed:
Added file read function (to refine save function later)

alexandermula committed:
Store dates as date object

alexandermula committed:
Merge branch 'branch-Level-8'

Kennethsim2000 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,48 @@
+public class Deadline extends Task {
+    private String description;
+    private LocalDate date;
+    private boolean isDone;
```
Good naming of booleans:)

Kennethsim2000 commented on src/main/java/Chacha.java:
```diff
@@ -0,0 +1,103 @@
+    public static void main(String[] args) {
+        System.out.println("Hello! I'm Chacha\n" + "What can I do for you?");
+        Scanner input = new Scanner(System.in);
+        String s = input.nextLine();
```
Maybe can name s differently? To a clearer variable

Kennethsim2000 commented on src/main/java/Chacha.java:
```diff
@@ -0,0 +1,103 @@
+        System.out.println("Hello! I'm Chacha\n" + "What can I do for you?");
+        Scanner input = new Scanner(System.in);
+        String s = input.nextLine();
+        ArrayList<Task> taskList = new ArrayList<Task>();
```
Good that the variable names are in camel case

Kennethsim2000 reviewed (commented):
Ultimately, good job on the code!

alexandermula committed:
Organise classes into suitable packages

alexandermula committed:
abstract read user input to ui class

alexandermula committed:
add list command

alexandermula committed:
add unmark command

alexandermula committed:
add all commands

alexandermula committed:
add parser logic

alexandermula committed:
fix parser logic

alexandermula committed:
add TaskList class

alexandermula committed:
add ui logic

alexandermula committed:
fix storage logic

alexandermula committed:
clean up packages
add storage logic
fix parser

alexandermula committed:
Merge branch 'nus-cs2103-AY2223S1:add-gradle-support' into add-gradle-support

alexandermula committed:
add gradle support

hoang227 commented on src/main/java/chacha/Storage.java:
```diff
@@ -0,0 +1,102 @@
+package chacha;
+
+import java.io.BufferedWriter;
```
Maybe you should consider adding spaces in between import statements from different packages to improve readability.

alexandermula committed:
add junit tests

Gavzzz commented on src/main/java/Chacha.java:
```diff
@@ -89,4 +91,13 @@ public static void main(String[] args) {
+            //load task objects into array
+            System.out.println(s.nextLine());
+        }
+    }
```
Overall code looks clean and should have no problem finishing up the ip

hoang227 commented on src/main/java/chacha/Storage.java:
```diff
@@ -0,0 +1,102 @@
+            dateTime = LocalDateTime.parse(date, formatter);
+        }  
+        Task task = new Task();
+        switch (type) {
```
The indentation for the switch statement does not follow the recommended standard. 'switch' and 'case' should have the same indentation.
![switch statement](https://user-images.githubusercontent.com/83692822/188309062-6ac321d6-6332-4310-b505-0a3bd6a48adc.png)

Gavzzz commented on src/main/java/Chacha.java:
```diff
@@ -6,15 +6,43 @@ public static void main(String[] args) {
         Scanner input = new Scanner(System.in);
         String s = input.nextLine();
-        ArrayList<String> list = new ArrayList<String>();
+        ArrayList<Task> taskList = new ArrayList<Task>();
```
Can consider making taskList a private static final field in the class

Gavzzz commented on src/main/java/Chacha.java:
```diff
@@ -6,15 +6,43 @@ public static void main(String[] args) {
+                        t.getDescription()); 		
 	            }   
+                 
+            } else if (s.contains("unmark")) {
```
Can consider using s.startsWith function to check if the task type is indeed the first input

alexandermula committed:
Add javadocs for TaskList

alexandermula committed:
Add javadocs to all classes

alexandermula committed:
Fix code to comply with coding standard

alexandermula committed:
Add find function

alexandermula committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/chacha/TaskList.java
#	src/main/java/chacha/commands/ExitCommand.java
#	src/main/java/chacha/commands/ListCommand.java
#	src/main/java/chacha/tasks/Deadline.java
#	src/main/java/chacha/tasks/Todo.java

alexandermula committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/chacha/Ui.java

alexandermula committed:
Add checkstyle

alexandermula committed:
Add view and image resources

alexandermula committed:
Add javafx

alexandermula committed:
add javaFX

alexandermula committed:
add assert statements

alexandermula committed:
Merge pull request #2 from alexandermula/branch-A-Assertions

add assert statements

alexandermula committed:
fix code quality

alexandermula committed:
Merge pull request #3 from alexandermula/branch-A-CodeQuality

fix code quality

alexandermula committed:
Add sort command

Sort command is null safe to handle todo tasks.

A sort is necessary for users to sort tasks chronologically.

Using a custom comparator class for tasks is preferable because other sorting parameters such as description can be added later.

Refer to this S/O discussion on dealing with multiple comparators as a chain
https://codereview.stackexchange.com/questions/229497/handle-null-in-comparator-class

alexandermula committed:
Merge pull request #4 from alexandermula/branch-C-Sort

Add sort command

alexandermula committed:
Find command: add varargs to pass multiple keywords

Find command finds returns all tasks with any of the keywords.

A find command which accepts multiple arguments is more user-friendly for users who wish to search for multiple types of tasks at once.

alexandermula committed:
Merge branch 'master' of https://github.com/alexandermula/ip

alexandermula committed:
Merge pull request #5 from alexandermula/branch-C-Sort

Find command: add varargs to pass multiple keywords

alexandermula committed:
Merge branch 'master' of https://github.com/alexandermula/ip

alexandermula committed:
Add enums

Enums control the switch statement in parser.

A separate enum class will allow for ease of future additions of commands.

alexandermula committed:
Add styling to GUI

Edit the fxml view files to update styling for GUI.

Styling the GUI makes the chatbot more user friendly.

alexandermula committed:
Add personality

Modify UI responses and error responses to create personality for Chacha.

Adding a personality improves user-friendliness of Chacha's interactions with users.

alexandermula committed:
Add github page

Github page as documentation for user.

By adding a website, users can easily refer to usage of the product.

alexandermula committed:
Build jar file

alexandermula committed:
Replace tabs usage with spaces

alexandermula committed:
Fix header comments

alexandermula committed:
Remove dead code

alexandermula committed:
Fix package name style

alexandermula committed:
Add test cases

alexandermula committed:
Fix header comments

alexandermula committed:
Fix code quality

-Fix arrowhead in Chacha getResponse

alexandermula committed:
Fix code quality

-Fix Parser SLAP
